### DISCUSSION TITLE
CLI tool with service-token is not returning results

### DISCUSSION THREAD

**Discussion by mohsen1** [planetscale/discussion#267] [Bug Reports] [answered]
When I specify `--service-token` I get the `"No databases have been created yet."` even though I have databases in this org:

```
pscale db list --org azimi --service-token pscale_tkn_OMTTED --service-token-id vyst07jboqne
No databases have been created yet.
```

But without those arguments (when I'm logged in):

```
pscale org show
azimi (from file: /Users/mohsen_azimi/.config/planetscale/pscale.yml)
```
```
pscale db list   
  NAME               CREATED AT   UPDATED AT   NOTES  
 ------------------ ------------ ------------ ------- 
  my-db-name   2 days ago   1 day ago           
```

**Comment by mohsen1** [comment, marked as answer]
my account has two organizations. I wonder if that's why?

**Reply by jgreet** [reply]
That's correct. Service tokens are scoped to a single organization.

**Reply by mohsen1** [reply]
I am specifying `--org` and I am 100% sure this service token is for the correct organization

**Reply by jgreet** [reply]
Sorry, I misread the question. 

Have you added an access grant to the token? They don't start with any and can't "see" a database by default.

New token:

```sh
$ pscale service-token create
  ID             TOKEN
 -------------- --------------------------------------------------------
  6hnhqpmd0fv7   pscale_tkn_xxx

$ pscale db list --service-token pscale_tkn_xxx --service-token-id 6hnhqpmd0fv7
No databases have been created yet.
```

Add an access grant for a database:
```sh
$ pscale service-token add-access 6hnhqpmd0fv7 read_branch --database beam
  DATABASE   ACCESSES
 ---------- -------------
  beam       read_branch

$ pscale db list --service-token pscale_tkn_xxx --service-token-id 6hnhqpmd0fv7
  NAME   CREATED AT     UPDATED AT   NOTES
 ------ -------------- ------------ -------
  beam   4 months ago   1 week ago
```

**Reply by mohsen1** [reply]
Thank you! This was defiantly an access issue. I had to create a new key and give it the right access rights.

I feel the CLI tool should show an "Access denied" error instead of just saying the list of databases is empty.

**Reply by jgreet** [reply]
Agreed, that error message is confusing in this situation. Opened https://github.com/planetscale/cli/issues/556
